CoffeeScript React component generator is fixed

ViXP opened the pull request:
CoffeeScript React component generator is fixed to modules export of component and to require the React object.
`rails g react:component Sample --coffee`
Before:
```
class @Sample extends React.Component
  render: ->
    `<div />`
```
After:
```
React = require('react')

class @Sample extends React.Component
  render: ->
    `<div />`

module.exports = @Sample
```

BookOfGreg commented on lib/generators/react/component_generator.rb:
```diff
@@ -125,15 +125,23 @@ def component_name
+          if options[:coffee]
+            %|React = require('react')\n|
+          else
+            %|var React = require("react")\n|
```
Would these work as import rather than requires if it's webpack?

BookOfGreg commented on lib/generators/react/component_generator.rb:
```diff
@@ -125,15 +125,23 @@ def component_name
+          if options[:coffee]            
+            %|module.exports = @#{component_name}|
+          else
+            %|module.exports = #{component_name}|
```
Similar to above, could these be export default rather than module.exports?

ViXP commented:
Yeah, I've corrected the code you mentioned, it is right to stick to one of the possible syntaxes. Also, PropTypes was removed from base 'react' source code, so now it must be imported in such way: `import PropTypes from 'prop-types'`. I've added this line to the generator too.

BookOfGreg commented:
I've merged your code and commits into #798 and corrected the associated tests. Thank you for your help with Coffeescript! Could you test out that branch and see if it meets your needs?
